drivers/docker/stats_test.go

```diff
@@ -0,0 +1,47 @@
+package docker
+
+import (
+	"testing"
+	"time"
+
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/hashicorp/nomad/plugins/drivers"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDriver_DockerStatsCollector(t *testing.T) {
+	require := require.New(t)
+	src := make(chan *docker.Stats)
+	defer close(src)
+	dst := make(chan *drivers.TaskResourceUsage)
+	defer close(dst)
+	stats := &docker.Stats{}
+	stats.CPUStats.ThrottlingData.Periods = 10
+	stats.CPUStats.ThrottlingData.ThrottledPeriods = 10
+	stats.CPUStats.ThrottlingData.ThrottledTime = 10
+
+	stats.MemoryStats.Stats.Rss = 6537216
+	stats.MemoryStats.Stats.Cache = 1234
+	stats.MemoryStats.Stats.Swap = 0
+	stats.MemoryStats.MaxUsage = 6651904
+
+	go dockerStatsCollector(dst, src, time.Second)
+
+	select {
+	case src <- stats:
+	case <-time.After(time.Second):
+		require.Fail("sending stats should not block here")
```

Comment:
TIL there's a require.Fail func! I've still been using t.Fatal in these cases but will stop now. :man_facepalming: